tasks/insertionTasks.ts

```diff
@@ -0,0 +1,235 @@
+            console.log(`##[debug]${error.stack}`);
+        }
+        throw error;
+    }
+});
+
+async function extractRoslynVersionFromManifest(manifestPath: string): Promise<string | null> {
+    const xmlFile = path.join(manifestPath, 'OfficialBuild.xml');
+
+    if (!fs.existsSync(xmlFile)) {
+        logError(`OfficialBuild.xml not found at ${xmlFile}`);
+        return null;
+    }
+
+    const xmlContent = fs.readFileSync(xmlFile, 'utf8');
+    const parser = new xml2js.Parser();
+    const result = await parser.parseStringPromise(xmlContent);
+
+    const packages = result?.Build?.Package || [];
+    for (const pkg of packages) {
+        const attrs = pkg.$;
+        if (attrs?.Id === 'Microsoft.CodeAnalysis.Common') {
+            return attrs.Version;
+        }
+    }
+
+    return null;
+}
+
+async function verifyRoslynRepo(roslynRepoPath: string): Promise<void> {
+    if (!fs.existsSync(roslynRepoPath)) {
+        throw new Error(`Roslyn repository not found at ${roslynRepoPath}`);
+    }
+    console.log(`Using Roslyn repository at ${roslynRepoPath}`);
+}
+
+async function generatePRList(startSHA: string, endSHA: string, roslynRepoPath: string, options: InsertionOptions): Promise<string> {
+    console.log(`Generating PR list from ${startSHA} to ${endSHA}...`);
+
+    try {
+        const { stdout } = await execAsync(
+            `cd "${roslynRepoPath}" && roslyn-tools pr-finder -s "${startSHA}" -e "${endSHA}" --format changelog --label VSCode`,
+            { maxBuffer: 10 * 1024 * 1024 } // 10MB buffer
+        );
+        return stdout || '(failed to generate PR list, see pipeline for details)';
+    } catch (error) {
+        const errorMessage = error instanceof Error ? error.message : String(error);
+        logWarning(`PR finder failed, using empty list: ${errorMessage}`);
+        if (error instanceof Error && error.stack) {
+            console.log(`##[debug]${error.stack}`);
+        }
+        return '(failed to generate PR list, see pipeline for details)';
+    }
+}
+
+async function updatePackageJson(newVersion: string): Promise<void> {
+    console.log(`Updating package.json with Roslyn version ${newVersion}...`);
+    const packageJsonPath = 'package.json';
+    const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'));
+
+    if (!packageJson.defaults) {
+        throw new Error('Could not find defaults section in package.json');
+    }
+    packageJson.defaults.roslyn = newVersion;
+
+    fs.writeFileSync(packageJsonPath, JSON.stringify(packageJson, null, 2) + '\n');
+}
+
+async function updateChangelog(version: string, prList: string, buildNumber?: string, buildId?: string): Promise<void> {
+    console.log('Updating CHANGELOG.md...');
+    const changelogPath = 'CHANGELOG.md';
+    const text = fs.readFileSync(changelogPath, 'utf8');
+    const NL = os.EOL;
+
+    // Prepare PR list (filter out 'View Complete Diff' lines)
+    const prLines = prList
+        ? prList
```

Comment:
as far as I can tell `prList` is never empty - probably need to remove these conditionals